micropub.php

```diff
@@ -114,17 +110,94 @@ public static function parse_query( $wp ) {
 		else {
 			$user_id = Micropub::authorize();
 		}
+		// If q is set this must be a query so 
+		if ( isset( $_GET['q'] ) ) {
+			Micropub::query_handler( $user_id );  // calls wp_die
+			wp_die();
+		}
+		// This handles the form encoded requests.
+		// TODO: Rewrite Form Handler and add JSON Handler
+		self::form_handler( $user_id );
+		// The handlers should exit before ever hitting the wp_die
+		wp_die();
+	}
+
+	/**
+	 * Use tokens.indieauth.com to validate the access token.
+	 *
+	 * If the token is valid, returns the user id to use as the post's author, or
+	 * NULL if the token only matched the site URL and no specific user.
+	 */
+	private static function authorize() {
```

Comment:
just fyi, this PR's diff is misleading and looks a lot bigger than it is because you moved this function in the file. if you leave it where it is instead, after `form_handler`, the diff is way smaller and more readable, and better represents the actual changes, which makes the PR easier to understand and review, and also more readable later.

not a big deal, no need to change it now. just a heads up for the future.